javascript/2017-grid-game.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * https://leetcode.com/problems/grid-game/
+ * Linear
+ * Time O(n) | Space O(n)
+ * @param {number[][]} grid
+ * @return {number}
+ */
+var gridGame = function(grid) {
+
+    const preSum1 = [0];
```

Comment:
Suggestion: break up into modular functions

```js
main = () => {
    preCompute()

    return search()
}
```

Reply:
I have split them into modules please have a look.

Reply:
Just a kind reminder that I updated the code as suggested.